test/functional/command/test_install_deploy.py

```diff
@@ -511,6 +511,45 @@ def package_info(self):
         expected = sorted(["pkga.so", "pkgb.so", "pkga.dll"])
         assert sorted(os.listdir(os.path.join(c.current_folder, "myruntime"))) == expected
 
+# This test requires in Windows to have symlinks enabled, otherwise it will fail
+@pytest.mark.parametrize("symlink, expected",
+                         [(True, ["libfoo.so.0.1.0", "libfoo.so.0", "libfoo.so"]),
+                          (False, ["libfoo.so.0.1.0",])])
+def test_runtime_deploy_symlinks(symlink, expected):
+    """ The deployer runtime_deploy should preserve symlinks when deploying shared libraries
+    """
+    c = TestClient()
+    conanfile = textwrap.dedent("""
+           from conan import ConanFile
+           from conan.tools.files import copy, chdir
+           import os
+           class Pkg(ConanFile):
+               package_type = "shared-library"
+               def package(self):
+                   copy(self, "*.so*", src=self.build_folder, dst=self.package_folder)
+                   with chdir(self, os.path.join(self.package_folder, "lib")):
+                       os.symlink(src="libfoo.so.0.1.0", dst="libfoo.so.0")
+                       os.symlink(src="libfoo.so.0", dst="libfoo.so")
+           """)
+    c.save({"foo/conanfile.py": conanfile,
+            "foo/lib/libfoo.so.0.1.0": "",})
+    c.run("export-pkg foo/ --name=foo --version=0.1.0")
+    c.run(f"install --requires=foo/0.1.0 --deployer=runtime_deploy --deployer-folder=output -c:a tools.deployer:symlinks={symlink}")
+
+    sorted_expected = sorted(expected)
+    assert sorted(os.listdir(os.path.join(c.current_folder, "output"))) == sorted_expected
+    if symlink:
```

Comment:
When `symlink` is False, should we add another assert line?

Reply:
Sorry, I don't follow, what should tested in case symlink is False?

Reply:
Ah do you mean about the windows error. Right, it's what Memsharded before. I'll skip it, good catch!

Reply:
Done [481578e](https://github.com/conan-io/conan/pull/17824/commits/481578ede0187e7781aec21dc95241a1c6c83c3b)